### PR TITLE
sys/net/gnrc_netif2: port sx127x to netif2

### PR DESCRIPTION
**Pull request by aabadie**
This is an attempt to port the SX127x driver to gnrc_netdev and gnrc_netif2 in the mean time.

This PR is adaption of #7356 for netif2.

Building the default example for b-l072z-lrwan1 fails and I don't understand why. Any help would be highly appreciated.

### CODE REVIEW

File: sys/auto_init/netif/auto_init_sx127x.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        LOG_DEBUG("[auto_init_netif] initializing sx1276 #%u\n", i);
+#endif
+
+        sx127x_setup(&sx127x_devs[i], &sx127x_params[i]);
+#ifdef MODULE_GNRC_NETIF2
+        gnrc_netif2_raw_create(sx127x_stacks[i], SX127X_STACKSIZE, SX127X_PRIO,
+                               "sx127x", (netdev_t *)&sx127x_devs[i]);
+#else
+        gnrc_netdev_init(sx127x_stacks[i], SX127X_STACKSIZE, SX127X_PRIO,
+                         "sx127x", &sx127x_netdevs[i]);
```
Comment (miri64) [inline sys/auto_init/netif/auto_init_sx127x.c]: This isn't necessary anymore, please remove all the `#ifdefs`

File: sys/auto_init/netif/auto_init_sx127x.c [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+#if defined(MODULE_SX1272)
+        LOG_DEBUG("[auto_init_netif] initializing sx1272 #%u\n", i);
+#else /* MODULE_SX1276 */
+        LOG_DEBUG("[auto_init_netif] initializing sx1276 #%u\n", i);
+#endif
+
+        sx127x_setup(&sx127x_devs[i], &sx127x_params[i]);
+#ifdef MODULE_GNRC_NETIF2
+        gnrc_netif2_raw_create(sx127x_stacks[i], SX127X_STACKSIZE, SX127X_PRIO,
+                               "sx127x", (netdev_t *)&sx127x_devs[i]);
```
Comment (miri64) [inline sys/auto_init/netif/auto_init_sx127x.c]: I don't think you want to use raw, but the LoRA layer. Otherwise you don't get any link-layer headers.

**Comment by aabadie** [comment]
> Please remove the gnrc_netdev related stuff

Done, I think.

File: boards/b-l072z-lrwan1/Makefile.dep [hunk 3]
```diff
@@ -1,3 +1,7 @@
+ifneq (,$(filter gnrc_netdev_default netdev_default,$(USEMODULE)))
```
Comment (miri64) [inline boards/b-l072z-lrwan1/Makefile.dep]: The `gnrc_netdev_default` case isn't required anymore, even in master ;-)

File: sys/auto_init/netif/auto_init_sx127x.c [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+ * @brief   Calculate the number of configured SX127x devices
+ */
+#define SX127X_NUMOF        (sizeof(sx127x_params) / sizeof(sx127x_params_t))
+
+/**
+ * @brief   Define stack parameters for the MAC layer thread
+ */
+#define SX127X_STACKSIZE           (THREAD_STACKSIZE_DEFAULT)
+#ifndef SX127X_PRIO
+#define SX127X_PRIO                (GNRC_NETDEV_MAC_PRIO)
```
Comment (miri64) [inline sys/auto_init/netif/auto_init_sx127x.c]: Please use `GNRC_NETIF2_MAC_PRIO` instead.

**Comment by aabadie** [comment]
@miri64, is it fine if I close this one and wait for the new gnrc netif to arrive in master ?

**Comment by miri64** [comment]
You can close it or adapt it as soon as it is in master. As you prefer :-).

**Comment by aabadie** [comment]
> You can close it or adapt it as soon as it is in master

Since it's targeting another branch than master, I think I'll have to close it in both cases. So let's do it now :)

**Comment by miri64** [comment]
(I know why, but for posterity) Why did you close this PR?

**Comment by aabadie** [comment]
> Why did you close this PR?

I think it's not possible to change the target branch in a PR on GitHub (but maybe I'm wrong).

**Comment by miri64** [comment]
It is. Just click on "Edit" next to the PR title above and change it ;-).

**Comment by aabadie** [comment]
> just click on "Edit" next to the PR title above and change it 

Ah ! Thanks for the tip :)
Anyway, I could make some cleanup in the initial branches.

**Comment by miri64** [comment]
What I actually wanted to see was: Superseded by #8158 and #8159 ;-)

**Pull request closed without merging**